packages/scene-composer/src/components/three-fiber/AnimationComponent/AnimationComponent.tsx

```diff
@@ -0,0 +1,81 @@
+// eslint-disable-next-line import/default
+import React, { Fragment, useEffect } from 'react';
+import * as THREE from 'three';
+import { useFrame } from '@react-three/fiber';
+
+import { useSceneComposerId } from '../../../common/sceneComposerIdContext';
+import { ISceneNodeInternal, useEditorState, useViewOptionState, IAnimationComponentInternal } from '../../../store';
+import { IAnimationComponent } from '../../../interfaces';
+
+const toggleIsAnimationPaused = (
+  object: THREE.Object3D,
+  AnimationMixer: THREE.AnimationMixer,
+  isPaused: boolean,
+  AnimationList: string[],
+) => {
+  try {
+    AnimationList.forEach(function (clip) {
+      const animation = THREE.AnimationClip.findByName(object.animations, clip);
+      AnimationMixer.clipAction(animation).paused = isPaused;
+    });
+  } catch (error) {
+    // console.error(error);
+  }
+};
+
+const list: any[] = [];
```

Comment:
This list variable doesn't read easily and gets confusing quickly. It looks like a component specific list, but each component pushes just one instance of a component, obj pair to it... for each time the useEffect gets triggered. The multiple instances get ignored by the getClip as it returns early.  But it seems like this should really be a map hold one entry per animation component and that it should be moved into react statement management to control updates to it and so that it has a clear owner.